include/merlin/utils.cuh

```diff
@@ -71,6 +71,13 @@ __inline__ __device__ uint64_t atomicAdd(uint64_t* address,
 namespace nv {
 namespace merlin {
 
+template <class S>
+static __forceinline__ __device__ S device_nano() {
+  S mclk;
+  asm volatile("mov.u64 %0,%%globaltimer;" : "=l"(mclk));
```

Comment:
Maybe add a test to compare the timestamp from timer on gpu and cpu.

Reply:
It's a little different, but I provide a `host_nano` for the host code to get the device timer in the test and benchmark util